zstacklib/zstacklib/storage/nbd/client.py

```
@@ -0,0 +1,292 @@
+        magic = struct.pack('>Q', NBD_OPTS_MAGIC)
+        opt = struct.pack('>L', NBD_OPT_EXPORT_NAME)
+        name_size = struct.pack('>L', len(name_bytes))
+
+        payload = magic + opt + name_size + name_bytes
+        sock.sendall(payload)
+
+        response = sock.recv(64)
+        if len(response) == 0:
+            raise NbdExportError(
+                f"Export selection failed. Export name '{name}' may be incorrect"
+            )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)